source/MaterialXRenderGlsl/GlslProgram.cpp

```diff
@@ -335,6 +335,12 @@ void GlslProgram::bindMesh(MeshPtr mesh)
         throw ExceptionRenderError("No mesh to bind");
     }
 
+    if (_boundMesh && mesh->getName() != _boundMesh->getName())
```

Comment:
@ashwinbhat In this test, perhaps we should really compare the two _shared pointers_, rather than comparing the names of the meshes?  Because shared pointer equality is based on the objects they point to, this should correctly handle cases that a name comparison cannot, e.g. switching between geometries with missing or duplicated name strings.